implementations/rust/daemon/src/args.rs

```diff
@@ -0,0 +1,127 @@
+/// running `ockamd` in control-mode.
+pub struct Config {}
+
+/// Command-line arguments passed to `ockamd`.
+#[derive(Debug, StructOpt)]
+#[structopt(
+    author = "Ockam Developers (ockam.io)",
+    about = "Encrypt and route messages using the Ockam daemon."
+)]
+pub struct Args {
+    #[structopt(
+        short,
+        long,
+        help = "Execute `ockamd` in control-mode, otherwise will start as a long-running process"
+    )]
+    control: bool,
+    #[structopt(
+        short = "p",
+        long = "port",
+        default_value = "11199",
+        help = "port for runtime configuration updates"
+    )]
+    control_port: u16,
+    #[structopt(
+        short,
+        long,
+        default_value = "stdin",
+        help = "data source providing input to `ockamd`"
+    )]
+    input: InputKind,
+    #[structopt(
+        short,
+        long,
+        help = "socket address of Ockam router, used when 'transport' option is set"
+    )]
+    output: Option<IpAddr>,
+    #[structopt(
+        short,
+        long,
+        default_value = "stdout",
+        help = "transport over which encrypted messages are sent"
+    )]
+    transport: OutputKind,
+    #[structopt(
+        parse(from_os_str),
+        long,
+        default_value = "ockamd.pub",
+        help = "path on disk to pre-existing public key"
+    )]
+    public_key: PathBuf,
+}
+
+impl FromStr for InputKind {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "stdin" => Ok(InputKind::Stdin),
+            _ => Err(unrecognized_input("input", s)),
+        }
+    }
+}
+
+impl FromStr for OutputKind {
+    type Err = String;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "stdout" => Ok(OutputKind::Stdout),
+            "udp" => Ok(OutputKind::Udp),
+            _ => Err(unrecognized_input("transport", s)),
+        }
+    }
+}
+
+fn unrecognized_input(flag: &str, input: &str) -> String {
+    format!("Unrecognized value ({:?}) for `{}` flag.", input, flag)
+}
+
+/// The mode in which `ockamd` is to be run.
+#[derive(Debug, StructOpt)]
+pub enum Mode {
+    /// Used for controlling configuration options at runtime, requiring that a
+    /// Server process of `ockamd` is running.
+    Control,
+    /// Used to create a long-running process, to be executed with a particular
+    /// input, e.g. Stdin.
+    Server,
+}
+
+#[derive(Debug, StructOpt)]
+pub enum InputKind {
+    Stdin,
+}
+
+#[derive(Debug, StructOpt)]
+pub enum OutputKind {
+    Stdout,
+    Udp,
+}
+
+impl Default for Args {
+    fn default() -> Self {
+        Self {
+            control: false,
+            control_port: DEFAULT_CONFIG_PORT,
+            input: InputKind::Stdin,
+            output: None, // TODO: should it be more literally named & exposed as such to the user, e.g. "onward_route"?
+            transport: OutputKind::Stdout,
+            public_key: PathBuf::from("ockamd.pub"),
```

Comment:
@mikelodder7 - with regard to the persistent vault, I think the minimal feature set is effectively the software vault plus a way to save the keys to disk, and load one from disk on initialization. @mrinalwadhwa suggested that we modify the initialization of a vault to optionally take bytes in to be used as the keys (Mrinal, check me on this please). 

The `public_key` field above in the `Args` struct may be incorrect. Would I need to load a key-pair path on disk in order to re-create a vault from a past initialization? If so, I'll update this field. Thanks.

Reply:
> modify the initialization of a vault to optionally take bytes in to be used as the keys

Here is what I was imagining
1. the persistent vault is initialized with a directory path.
2. It gets to manage private keys in that path.
3. All files stored in that dir are private keys.
4. The name of the file, is the identifier of that private key.
5. Public keys, I think, need not be stored because they can be derived from private keys.

The above is kinda like the SSH home folder.

Reply:
I can help with that @nilslice

Reply:
Thanks, @mikelodder7. I won't have time to sync up today -- but I'll find some time early next week to discuss. If you are in a position to start, don't let me hold you back :)